install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -650,6 +650,19 @@ enableK8sEventHandover: false
 # -- Enable CiliumEndpointSlice feature.
 enableCiliumEndpointSlice: false
 
+envoyConfig:
+  # -- Enable CiliumEnvoyConfig CRD
+  # CiliumEnvoyConfig CRD can also be implicitly enabled by other options.
+  enabled: false
```

Comment:
Bringing feedback from the other PR, the conventional way of indicating that a value has a runtime-computed default is to use nil / `~`.

This also allows the end-user to force-override the defaulting logic in either direction.